### PR TITLE
format and int generic for x86

### PR DESCRIPTION
**Pull request by matbesancon**
This should resolve the current fail on x86, hard-coded Int64 blocked conversions

### CODE REVIEW

**Comment by matbesancon** [comment]
We will also see if the error observed on nightly has been fixed

**Comment by matbesancon** [comment]
So COSMO.jl seems to be the cause of test failure on x86 architecture. And latest has some broken JLL artifacts.

File: src/conic_form.jl [hunk 1]
```diff
@@ -26,14 +26,14 @@ _set_type(::MOI.ConstraintIndex{F,S}) where {F,S} = S
 cons_offset(conic::MOI.Zeros) = conic.dimension
 cons_offset(conic::MOI.Nonnegatives) = conic.dimension
 cons_offset(conic::MOI.SecondOrderCone) = conic.dimension
-cons_offset(conic::MOI.PositiveSemidefiniteConeTriangle) = Int64((conic.side_dimension*(conic.side_dimension+1))/2)
+cons_offset(conic::MOI.PositiveSemidefiniteConeTriangle) = Int((conic.side_dimension*(conic.side_dimension+1))/2)
```
Comment (blegat) [inline src/conic_form.jl]: This should be `div(conic.side_dimension * (conic.side_dimension+1), 2)` or even better, we should replace this `con_offset` by `MOI.dimension`

Comment (matbesancon) [reply src/conic_form.jl]: I replaced the implementation with MOI.dimension. I didn't want to remove cons_offset altogether in the same PR, maybe do so in a follow-up

File: src/conic_form.jl [hunk 2]
```diff
@@ -26,14 +26,14 @@ _set_type(::MOI.ConstraintIndex{F,S}) where {F,S} = S
 cons_offset(conic::MOI.Zeros) = conic.dimension
 cons_offset(conic::MOI.Nonnegatives) = conic.dimension
 cons_offset(conic::MOI.SecondOrderCone) = conic.dimension
-cons_offset(conic::MOI.PositiveSemidefiniteConeTriangle) = Int64((conic.side_dimension*(conic.side_dimension+1))/2)
+cons_offset(conic::MOI.PositiveSemidefiniteConeTriangle) = MOI.dimension(conic)
```
Comment (blegat) [inline src/conic_form.jl:29]: All `cons_offset` method are equivalent to `MOI.dimension` so we should just remove this `cons_offset` function.

Comment (blegat) [reply src/conic_form.jl:29]: Missed your other comment, let's do it in a separate PR indeed

**Review by blegat — APPROVED** [review]

**Pull request merged**